src/dune_rules/lib_file_deps.ml

```diff
@@ -1,22 +1,31 @@
 open Import
 
 module Group = struct
-  type t =
+  type ocaml =
     | Cmi
     | Cmx
+
+  type melange = Melange.Cm_kind.t
+
+  type t =
+    | Ocaml of ocaml
+    | Melange of melange
     | Header
 
-  let all = [ Cmi; Cmx; Header ]
+  let all = [ Ocaml Cmi; Ocaml Cmx; Melange Cmi; Melange Cmj; Header ]
 
   let ext = function
-    | Cmi -> Cm_kind.ext Cmi
-    | Cmx -> Cm_kind.ext Cmx
+    | Ocaml Cmi -> Cm_kind.ext Cmi
+    | Ocaml Cmx -> Cm_kind.ext Cmx
+    | Melange Cmi -> Lib_mode.Cm_kind.ext (Melange Cmi)
+    | Melange Cmj -> Lib_mode.Cm_kind.ext (Melange Cmj)
     | Header -> Foreign_language.header_extension
 
   let obj_dir t obj_dir =
     match t with
-    | Cmi -> Obj_dir.public_cmi_dir obj_dir
-    | Cmx -> Obj_dir.native_dir obj_dir
+    | Ocaml Cmi | Melange Cmi -> Obj_dir.public_cmi_dir obj_dir
```

Comment:
Shouldn't this be the Melange cmi dir?

Reply:
I think so. Still unclear how public cmi dirs will work with melange but for now using melange_dir for all melange kinds (https://github.com/jchavarri/dune/pull/4/commits/9f2477cec08fb14998533a38e1df0731be058051).